swaybar/ipc.c

```diff
@@ -322,8 +335,11 @@ void ipc_bar_init(struct bar *bar, const char *bar_id) {
 	}
 	free(res);
 	json_object_put(outputs);
-
-	const char *subscribe_json = "[ \"workspace\", \"mode\" ]";
+	// Will need to be redone for display mode toggling compatibility
+	// Will require setting bar->config->hidden_state to "show" as default
+	const char *subscribe_json = strcmp(bar->config->display_mode, "hide") == 0 ?
+		"[ \"workspace\", \"mode\", \"modifier\" ]"
+		: "[ \"workspace\", \"mode\" ]";
```

Comment:
Wait, ok I see code in commands/bar/mode.c for handling bar display mode commands that sends an ipc barconfig event but the event never makes it to the socket because there's no security mapping there. I guess the question is whether to make bars subscribed to barconfig or bindings as a whole to do display mode switching.